Usage improvements

TheRaspPie opened the pull request:
<a href="https://github.com/SpongePowered/SpongeAPI/pull/834">Api PR</a> | Implementation PR

This pull request should improve the overall experience from the view of a plugin developer.
It should be merged before API Revision 3.0.

In summary, this pull request contains the tiny improvements that I think are needed to get into API Revision 3.0.

Please don't close this PR until Revision 3.0 as I may commit a few other changes in the meantime.

me4502 commented:
Can you please provide a quick summary of the changes, and the benefits that they bring?

TheRaspPie commented:
Of course.
The `Entity` interface in SpongeAPI had a bunch of convenience methods for teleporting:
```java
getWorld();
getLocation();
setLocation(Location);
setLocationSafely(Location);
getRotation();
setRotation(Vector3d);
setLocationAndRotation(Location, Vector3d);
setLocationAndRotationSafely(Location, Vector3d);
setLocationAndRotation(Location, Vector3d, EnumSet<RelativePositions> relativePositions);
setLocationAndRotationSafely(Location, Vector3d, EnumSet<RelativePositions> relativePositions);
transferToWorld(String worldName, Vector3d position);
transferToWorld(UUID uuid, Vector3d position);
```

I added these methods:
```java
getPosition();
setPosition(Vector3d);
setPositionSafely(Vector3d);
setPositionAndRotation(Vector3d, Vector3d);
setPositionAndRotationSafely(Vector3d, Vector3d);
setPositionAndRotation(Vector3d, Vector3d, EnumSet<RelativePositions> relativePositions);
setPositionAndRotationSafely(Vector3d, Vector3d, EnumSet<RelativePositions> relativePositions);
transferToWorld(String worldName, Vector3d position, Vector3d rotation);
transferToWorld(UUID uuid, Vector3d position, Vector3d rotation);
```

After talking to @Zidane he said that I should make a pull request for it and all other (minor) things I want to fix.

This Implementation PR only implements the new methods I mentioned in this post.
For more information about what I changed too, see the commits in Api PR.

gabizou commented:
Closing this as all of the changes in API need to be separate PR's.
